overlord/snapstate/snapstate.go

```diff
@@ -57,7 +57,7 @@ const (
 	// 0x40000000 >> iota
 )
 
-func doInstall(s *state.State, curActive bool, ss *SnapSetup) (*state.TaskSet, error) {
+func doInstall(s *state.State, curActive bool, snapst *SnapState, ss *SnapSetup) (*state.TaskSet, error) {
```

Comment:
doesn't make sense to pass curActive, if we pass the whole snapst afait, can just check snapst.Active then